File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Other/Internal/ViewSourceButtonHelper.vb
﻿Imports System.Collections.Generic
Imports System.Windows.Controls

Namespace OpenSilver.Samples.TelerikUI
    Friend Module ViewSourceButtonHelper
        Public Sub ViewSource(ByVal sourcePaths As List(Of ViewSourceButtonInfo))
            If sourcePaths.Count > 0 Then
                Dim tabControl = New TabControl()

                For Each viewSourceButtonInfo In sourcePaths
                    Dim tabItem = New TabItem() With {
    .Header = viewSourceButtonInfo.TabHeader,
                            .Content = New ControlToDisplayCodeHostedOnGitHub() With {
        .FilePathOnGitHub = viewSourceButtonInfo.FilePathOnGitHub
    }
}

                    tabControl.Items.Add(tabItem)
                Next

                CType(tabControl.Items(0), TabItem).IsSelected = True

                MainPage.Current.ViewSourceCode(tabControl)
            End If
        End Sub
    End Module
End Namespace

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Other/Internal/ViewSourceButtonHelper.cs
using System.Collections.Generic;
using Telerik.Windows.Controls;

namespace OpenSilver.Samples.TelerikUI
{
    static class ViewSourceButtonHelper
    {
        public static void ViewSource(List<ViewSourceButtonInfo> sourcePaths)
        {
            if (sourcePaths.Count > 0)
            {
                var tabControl = new RadTabControl();

                foreach (ViewSourceButtonInfo viewSourceButtonInfo in sourcePaths)
                {
                    var tabItem = new RadTabItem()
                    {
                        Header = viewSourceButtonInfo.TabHeader,
                        Content = new ControlToDisplayCodeHostedOnGitHub()
                        {
                            FilePathOnGitHub = viewSourceButtonInfo.FilePathOnGitHub
                        }
                    };

                    tabControl.Items.Add(tabItem);
                }

                ((RadTabItem)tabControl.Items[0]).IsSelected = true;

                MainPage.Current.ViewSourceCode(tabControl);
            }
        }
    }
}
